### ISSUE TITLE
typescript-react-apollo: allow document import (rather than using gql tag)

### ISSUE THREAD

**Issue by adam1658** [dotansimha/graphql-code-generator#1431]
**Is your feature request related to a problem? Please describe.**
I'd like the ability to provide the GraphQL Document myself, rather than having it copied in and used with the gql tag. Ordinarily I'd use parcel's built in graphql file support (which effectively gives me the JSON results of using graphql-tag at build time, rather than runtime)

**Describe the solution you'd like**
```js
// An `importDocument` option that replaces this:
export const Document = gql...``;
// with this:
import Document from './document-file.graphql';
export { Document };
```

**Describe alternatives you've considered**
Rather than have an `importDocument` option, there could be an option which justs disables the document generation, then I use another plugin to add the `import Document...` lines

**Additional context**
I'm using parcel to build my app (which has native support for importing graphql documents as their JSON representation)
Webpack users have the same ability by using the `graphql-tag/loader` webpack loader.

**Comment by dotansimha** [comment]
Hi @adam1658 !
Thank you for opening this issue. We are currently working on refactoring all typescript plugins (see PR: https://github.com/dotansimha/graphql-code-generator/pull/1353 ) 
We added the ability to use a JSON of the document (`DocumentNode` JSON), does it answer the performance requirements? or do you think the import from another file is the only way to achieve it?

**Comment by adam1658** [comment]
Including the document JSON rather than the document string would be ideal. Thanks @dotansimha

**Comment by dotansimha** [comment]
Fix available in 1.0.0 👍
